src/lambda/wizards/samDeployWizard.ts

```diff
@@ -524,11 +487,75 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
         })
     }
 
+    private async addBucketsToRegionMap(buckets: S3.Bucket[], s3Client: S3Client): Promise<void> {
+        const promises = buckets
+            .filter(bucket => bucket.Name)
+            .map(async bucket => {
+                const bucketLocation = await s3Client.getBucketLocation(bucket.Name!)
+                const transformedLocation = this.normalizeLocation(bucketLocation.LocationConstraint)
+                this.addBucketToRegionMap(bucket.Name!, transformedLocation)
+            })
+
+        await Promise.all(promises)
+    }
+
+    // correct blank and 'EU' locations:
+    // https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region
+    private normalizeLocation(location?: string): string {
+        if (!location || location === '') {
+            location = 'us-east-1'
+        } else if (location === 'EU') {
+            location = 'eu-west-1'
+        }
+
+        return location
+    }
+
     private addBucketToRegionMap(bucket: string, region: string) {
-        if (!this.regionBucketMap.has(region)) {
-            this.regionBucketMap.set(region, [])
+        if (!this.bucketsFilteredByRegion.has(region)) {
+            this.bucketsFilteredByRegion.set(region, [])
+        }
+        this.bucketsFilteredByRegion.get(region)!.push(bucket)
+    }
+
+    private createBucketQuickPickItems(
+        selectedRegion: string,
+        noBucketsMessage: string,
```

Comment:
It's awkward passing in `noBucketsMessage` and rolling the description in this method.

Choose one:
* don't pass in `noBucketsMessage`, create it in here when it is needed
* make the caller responsible for creating the "fallback item" if there are no buckets. Pass in a QuickPickItem to use when there is nothing else to add to it.

Reply:
✔️